gcovr/utils.py

```diff
@@ -228,3 +228,43 @@ def build_filter(regex):
         return re.compile(re.escape(os.getcwd() + "\\") + regex)
     else:
         return re.compile(os.path.realpath(regex))
+
+
+class Logger(object):
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+    def warn(self, pattern, *args, **kwargs):
+        """Write a formatted warning to STDERR.
+
+        pattern: a str.format pattern
+        args, kwargs: str.format arguments
+        """
+        pattern = "(WARNING) " + pattern + "\n"
+        sys.stderr.write(pattern.format(*args, **kwargs))
+
+    def error(self, pattern, *args, **kwargs):
+        """Write a formatted error to STDERR.
+
+        pattern: a str.format pattern
+        args, kwargs: str.format parameters
+        """
+        pattern = "(ERROR) " + pattern + "\n"
+        sys.stderr.write(pattern.format(*args, **kwargs))
+
+    def msg(self, pattern, *args, **kwargs):
+        """Write a formatted message to STDOUT.
+
+        pattern: a str.format pattern
+        args, kwargs: str.format arguments
+        """
+        pattern = pattern + "\n"
+        sys.stdout.write(pattern.format(*args, **kwargs))
```

Comment:
if no output file has been specified then the coverage output will be output on stdout as well. Is that a problem ?

Reply:
I just extracted the existing `sys.stderr.write()` and `print()` calls into this class. So each message continues to go were it went previously: warnings/errors to stderr, and --verbose messages to stdout.

Arguably that's wrong (for exactly your reason) and should be redesigned. However, this PR already has more than enough changes … (sorry about that!)

Reply:
If there are no regressions, that's fine by me. An issue shall probably be opened regarding this, and, as you said, this is another topic.